cmake/Modules/FindComputeCpp.cmake

```diff
@@ -494,12 +494,39 @@ function(add_sycl_to_target)
   # This has the same effect as:
   # target_link_libraries(${SNN_ADD_SYCL_TARGET}
   #   PUBLIC -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  # )
-  set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
-    LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  )
-  set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  )
+  # )  
+  if (MSVC)
+    # Allows compute++ to parse the Windows headers
+    list(APPEND COMPUTECPP_USER_FLAGS "-D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH=1")
+    if (MSVC_VERSION GREATER_EQUAL 1910)
+      # If /Zc:__cplusplus is not specified the compiler won't change the
+      # value of the macro __cplusplus. Lots of existing code rely on the
+      # value of this macro to determine whether or not some features are
+      # available. Setting this option will make sure the macro gets the
+      # correct value in Visual Studio versions 2017 and newer.
+      # https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus
+      set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
+        COMPILE_OPTIONS /Zc:__cplusplus
+      )
+    endif()
+    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
+      LINK_OPTIONS /FORCE
+    )
+    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
+      INTERFACE_LINK_OPTIONS /FORCE
+    )
+    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY 
+        LINK_LIBRARIES ComputeCpp::ComputeCpp
+    )
+    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
+      INTERFACE_LINK_LIBRARIES ComputeCpp::ComputeCpp
+    )
+  else()
+    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
+      LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
```

Comment:
Rather than having 
```
if (MSVC)
    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY 
        LINK_LIBRARIES ComputeCpp::ComputeCpp
    )
else()
    set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
      LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
    )
endif()
```
You could have something like
```
if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU|Clang")                                                            
  set(_platform_link_flags -Wl,--allow-shlib-undefined)
endif() 
set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY 
    LINK_LIBRARIES ${_platform_link_flags} ComputeCpp::ComputeCpp
)
```
Which makes it a bit more obvious what the difference is, and is more easily extensible in the future if needed.

Reply:
@mcleary this doesn't seem to be changed in the most recent version - please let me know if I'm mistaken!

Reply:
It was changed: https://github.com/codeplaysoftware/SYCL-DNN/pull/5/files

Reply:
Ok, I misread the suggestion. I applied the suggestion now.